h2o-py/h2o/frame.py

```diff
@@ -3776,7 +3776,14 @@ def hist(self, breaks="sturges", plot=True, **kwargs):
             try:
                 import matplotlib
                 if server:
-                    matplotlib.use("Agg")
+                    # Matplotlib version 3.1 and higher deprecates/doesn't have `warn` parameter
+                    version = matplotlib.__version__.split(".")
+                    if int(version[0]) < 3 or \
+                            (int(version[0]) == 3 and
+                             int(version[1]) < 1):
+                        matplotlib.use("Agg", warn=False)
+                    else:
+                        matplotlib.use("Agg")
```

Comment:
do we always use the same pattern when loading `matplotlib`? Can't we externalize this in a separate function?

Reply:
also for fun:
```
from distutils.version import LooseVersion as lv

if lv(version) < lv("3.1"):
    ...
```

Reply:
btw, you  write:
> Newer versions (3.2+) of matplotlib don't have the warn parameter

so it looks like the comparison operator should be `<=` instead of `<`

Reply:
> The solution would be prettier if it would be contained in a simple function, any ideas where to put it?

yeah, I agree.
The only module with similar function is probably `h2o.utils.shared_utils`: there's already way to many things there but well... it also contains functions like `can_use_pandas`.

Maybe it's worth creating a `h2o.utils.imports` module (or `h2o.utils.ext_dependencies`)  and move all those external modules detection logic + custom imports there.

Reply:
I think the original code is already a copy&paste, IMHO all we need to do is to

take `_get_matplotlib_pyplot` from `ModelBase`, put it in a utility class and use everywhere.

WDYT?

Reply:
Thanks for the suggestions! I looked into using `warnings` related functions to suppress it (something that @michalkurka suggested on slack) and there is couple issues with that, namely:

- adding a filter is modifying a global state, that can cause some issues when working with threads/asyncio
- some packages modify the warnings filter (that could override the filter that we would setup) or reroute the warning to logger (which still causes it to be shown in, e.g., Jupyter) (e.g. some versions of tensorflow)
- the warning messages differ between versions (at least between 2.x.x and 3.x.x) so I would either suppress all `UserWarnings` or go through the versions to find out if and in what they differ.

and while we could use `warnings.catch_warnings` context manager to make the filter more local, it would still have undefined behavior when using with multiple threads.

So I think the best approach is not to emit the warning at all by detecting the version and using the appropriate arguments.

Reply:
> and while we could use warnings.catch_warnings context manager to make the filter more local, it would still have undefined behavior when using with multiple threads.

maybe you're overthinking this, aren't you? I don't see many use cases where users would use multithread for plotting, doesn't sound reasonable, and it's definitely not the case in Jupyter/ipython environment.

Reply:
I guess I am. So you think suppressing the warning using filter would be a better solution?

(The warning is emitted when providing `server=True` where I guess it is a bit more likely to have multiple threads than just during regular plotting, but you are right that it would be a very unlikely scenario.)

Reply:
if the warning is emitted exactly when calling `matplotlib.use(...)`, then the current solution looks good to me.

The context manager approach is a bit cleaner, but I don't have a strong preference here. I'd validate both.